File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.java

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {
	static int n;
	static int m;
	static int[] arr;
	static int[] preSum;
	
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringBuilder sb = new StringBuilder();
		
		StringTokenizer stk = new StringTokenizer(br.readLine(), " ");
		n = Integer.parseInt(stk.nextToken());
		m = Integer.parseInt(stk.nextToken());
		arr = new int[n];
		preSum = new int[n+1];
		arr = Arrays.stream(br.readLine().split(" "))
				.mapToInt(x -> Integer.parseInt(x))
				.toArray();
		
		preSum[1] = arr[0];
		
		for(int i=1; i<n; i++) {
			preSum[i+1] += preSum[i] + arr[i]; 
		}
		
		// System.out.println(Arrays.toString(preSum));
		
		for(int t=0; t<m; t++) { // from to
			stk = new StringTokenizer(br.readLine(), " ");
			int from = Integer.parseInt(stk.nextToken());
			int to = Integer.parseInt(stk.nextToken());
			
			int result = preSum[to] - preSum[from-1]; 
			sb.append(result+ "\n");
		}
		
		System.out.print(sb.toString());
	}
}

File: 백준/Silver/11659. 구간 합 구하기 4/구간 합 구하기 4.cc
#include <stdio.h>

int arr[100005];

int main(){
    int n,m;
    int a,b;

    scanf("%d %d",&n,&m);

    for(int i=1;i<=n;i++){
        scanf("%d",&arr[i]);
        arr[i] += arr[i-1];
    }

    for(int i=0;i<m;i++){
        scanf("%d %d",&a,&b);

        printf("%d\n",arr[b]-arr[a-1]);
    }

    return 0;
}